src/main/java/com/spring/test/controller/UserController.java

```diff
@@ -0,0 +1,48 @@
+package com.spring.test.controller;
+
+import com.spring.test.dto.UserResponseDto;
+import com.spring.test.model.User;
+import com.spring.test.service.UserService;
+import java.util.List;
+import java.util.stream.Collectors;
+import org.modelmapper.ModelMapper;
+import org.springframework.stereotype.Controller;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RestController;
+
+@RestController
+@RequestMapping("/users")
+@Controller
+public class UserController {
+    private final UserService userService;
+
+    private final ModelMapper modelMapper;
```

Comment:
I don't whether it is critical or not, but checklist says to create a separate private method on the controller layer to map User to DTO.

Reply:
it seems to me that this mapper is the better solution